controllers/app.controller.js

```diff
@@ -12,8 +13,11 @@ exports.getTopics = (req, res) => {
     })
 };
 
-exports.getArticlesById = (req, res) => {
-    findArticlesById(req.params.article_id).then((data) => {
+exports.getArticlesById = (req, res, next) => {
+    findArticlesById(req.params.article_id)
+    .then((data) => {
+        if (data.rows.length === 0) throw new Error('404');
```

Comment:
Your error handling should be done in the model for this. Is it the models jobs to interact with the database and pass back to the controller the information it needs. 
You should also be using a Promise.reject here instead of a throw new Error. The fact that you're throwing a message of "404" then in the error handling middlewear you're checking that message to then give the error status and message is quite confusing.